modules/tinymce/src/core/main/ts/text/CharType.ts

```diff
@@ -13,7 +13,7 @@ const isNbsp = is(Unicode.nbsp);
 
 const isWhiteSpace = (chr: string) => chr !== '' && ' \f\n\r\t\v'.indexOf(chr) !== -1;
 
-const isContent = (chr: string) => !isWhiteSpace(chr) && !isNbsp(chr);
+const isContent = (chr: string) => !isWhiteSpace(chr) && !isNbsp(chr) && !Unicode.isZwsp(chr);
```

Comment:
I'm not entirely sure this is the correct solution, but it is a solution, and the tests work.
Short of it, there's a ZWSP added to the node which was counted as 'content'. It's removed some time after. The ZWSP may be added by cursor placement.

Reply:
Yes, that seems correct since zwnbsp are getting trimmed away when getting the content out we should never treat that as content.